### PR TITLE
Add recording directories

### PR DESCRIPTION
**Pull request by emveepee**
Backend API change allows passing recording directory name on pending shows.

### CODE REVIEW

**Comment by emveepee** [comment]
@ksoo @phunkyfish testing this I noticed that in-progress recordings allow the Recording group and Start padding to be changed.  Is this expected?  The backend does allow the End padding to be changed so an update should be allowed but I was wondering if I could disable these other two options without creating two new timer types

image

File: src/Timers.cpp [hunk 1]
```diff
@@ -319,6 +319,23 @@ bool Timers::UpdatePvrTimer(tinyxml2::XMLNode* pRecordingNode, kodi::addon::PVRT
     }
   }
 
+  if (status == "Pending")
+  {
+    std::string directory;
+    if (XMLUtils::GetString(pRecordingNode, "directory", directory))
+    {
+        int i = 0;
+        for (auto it = m_settings.m_recordingDirectories.begin(); it != m_settings.m_recordingDirectories.end(); ++it, i++)
```
Comment (AlwinEsch) [inline src/Timers.cpp]: As a question of understanding, is there a background to using the iterator and an additional "i" here and on other places?

Could here the following not enough?:
```cpp
  for (size_t i = 0; i < m_settings.m_recordingDirectories.size(); ++i)
```

Comment (emveepee) [reply src/Timers.cpp]: I took existing code https://github.com/kodi-pvr/pvr.nextpvr/blob/Matrix/src/Timers.cpp#L195b for a similar recording group lookup and reused it for this loop.   It dates back to Jarvis, I will change both for readability.

Comment (emveepee) [reply src/Timers.cpp]: @AlwinEsch I made the changes but used unsigned int rather then size_t to avoid compiler warning.

Comment (AlwinEsch) [reply src/Timers.cpp]: No problem, thought only about the size() call, but you are right, the size_t becomes then more in conflict on some other places where then bring warnings.

**Comment by emveepee** [comment]
> 
> @ksoo @phunkyfish testing this I noticed that in-progress recordings allow the Recording group and Start padding to be changed. Is this expected? The backend does allow the End padding to be changed so an update should be allowed but I was wondering if I could disable these other two options without creating two new timer types
> 

I stepped through SetTimer and it seems this is expected behaviour so I created a new timer type which does allow removing these options unfortunately it still gives a user the option of changing the timer type on an in-progress recording which is impossible so I might as just leave it as is.

**Comment by emveepee** [comment]
@phunkyfish  @sub3 although we missed the 19.1 window for packaged binaries, do I need to do anything else for this to be reviewed?

**Review by phunkyfish — APPROVED** [review]
Sorry, missed this.

**Pull request merged**